hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/lock/LockManager.java

```diff
@@ -186,21 +186,42 @@ private ActiveLock getLockForLocking(final R resource) {
      * be removed from the activeLocks map and returned to
      * the object pool.
      */
-    return activeLocks.compute(resource, (k, v) -> {
+    ActiveLock computedLock = activeLocks.compute(resource, (k, v) -> {
       final ActiveLock lock;
+      if (v == null) {
+        return null;
+      } else {
+        lock = v;
+      }
+      lock.incrementActiveCount();
+      return lock;
+    });
+
+
+    if (computedLock == null) {
       try {
-        if (v == null) {
-          lock = lockPool.borrowObject();
-        } else {
-          lock = v;
-        }
-        lock.incrementActiveCount();
+        final ActiveLock lock2 = lockPool.borrowObject();
+        computedLock = activeLocks.compute(resource, (k, v) -> {
+          final ActiveLock lock;
+          try {
+            if (v == null) {
+              lock = lock2;
+            } else {
+              lockPool.returnObject(lock2);
+              lock = v;
+            }
+            lock.incrementActiveCount();
+          } catch (Exception ex) {
+            throw new RuntimeException(ex);
```

Comment:
Nit: Is this necessary?  I don't see any checked exception being thrown in the `try` block, and unchecked exceptions don't need to be wrapped.

Reply:
Based on Xiaoyu's comment reverted this change.